--- src/symbolize/gimli/libs_aix.rs
+++ src/symbolize/gimli/libs_aix.rs
@@ -30,7 +30,7 @@
         loop {
             if libc::loadquery(
                 libc::L_GETINFO,
-                buffer.as_mut_ptr().cast::<libc::c_char>(),
+                buffer.as_mut_ptr().cast::<libc::c_void>(),
                 (mem::size_of::<libc::ld_info>() * buffer.len()) as u32,
             ) != -1
             {
